src/components/Tabs/AnnotationTab.tsx

```diff
@@ -0,0 +1,360 @@
+    return {
+      key: index.toString(),
+      label: label.ids.size ? `${label.name} (${label.ids.size})` : label.name,
+    };
+  });
+
+  // A small popup that appears when you press the edit button.
+  const editPopoverContents = (
+    <FlexColumn style={{ width: "250px" }} $gap={10}>
+      <label style={{ gap: "10px", display: "flex", flexDirection: "row" }}>
+        <span>Name</span>
+        <Input
+          value={editPopoverNameInput}
+          onChange={(e) => setEditPopoverNameInput(e.target.value)}
+          ref={editPopoverInputRef}
+        ></Input>
+      </label>
+      <label style={{ gap: "10px", display: "flex", flexDirection: "row" }}>
+        <span>Color</span>
+        <div>
+          <ColorPicker
+            size="small"
+            value={new AntdColor(selectedLabel?.color.getHexString() || "ff00ff")}
+            onChange={(_, hex) => {
+              currentLabelIdx !== null && setLabelColor(currentLabelIdx, new Color(hex as HexColorString));
+            }}
+            disabledAlpha={true}
+          />
+        </div>
+      </label>
+      <FlexRow style={{ marginLeft: "auto" }} $gap={10}>
+        <Button onClick={onClickEditCancel}>Cancel</Button>
+        <Button onClick={onClickEditSave} type="primary">
+          Save
+        </Button>
+      </FlexRow>
+    </FlexColumn>
+  );
+
+  return (
+    <FlexColumnAlignCenter $gap={10}>
+      {/* Top-level annotation edit toggle */}
+      <FlexRow style={{ width: "100%" }} $gap={6}>
+        <AnnotationModeButton
+          type="primary"
+          $color={isAnnotationModeEnabled ? "success" : "default"}
+          style={{ paddingLeft: "10px" }}
+          onClick={() => setIsAnnotationModeEnabled(!isAnnotationModeEnabled)}
+        >
+          {isAnnotationModeEnabled ? <CheckOutlined /> : <TagOutlined />}
+          {isAnnotationModeEnabled ? "Done editing" : "Create and edit"}
+        </AnnotationModeButton>
+        {isAnnotationModeEnabled && (
+          <p style={{ color: theme.color.text.hint }}>
+            <i>Editing in progress...</i>
+          </p>
+        )}
+      </FlexRow>
+
+      {/* Label selection and edit/create/delete buttons */}
+      <FlexRow $gap={10} style={{ width: "100%" }}>
+        <label style={{ gap: "5px" }}>
+          <SelectionDropdown
+            selected={(currentLabelIdx ?? -1).toString()}
+            items={selectLabelOptions}
+            onChange={function (key: string): void {
+              const index = parseInt(key, 10);
+              onSelectLabel(index);
+            }}
+            disabled={currentLabelIdx === null}
+            showTooltip={false}
+          ></SelectionDropdown>
+        </label>
+        <div>
+          {/* TODO: Remove color picker once color dots can be added to the dropdowns. */}
+          <ColorPicker
+            size="small"
+            value={new AntdColor(selectedLabel?.color.getHexString() || "ffffff")}
+            onChange={(_, hex) => {
```

Comment:
can you type `hex` here instead of coercing it below?

Reply:
I can't, since the expected type of `onChange` (which has `hex` as a string) is defined by `Ant.ColorPicker`.